### PR TITLE
Pull through Host.Storage and Extensions.Storage updates

### PR DESCRIPTION
**Pull request by paulbatum**
- Updates to storage SDK v11
- Includes a major version rev for WebJobs.Extensions

This would go in after https://github.com/Azure/azure-webjobs-sdk/pull/2356

### CODE REVIEW

File: src/WebJobs.Extensions/WebJobs.Extensions.csproj [hunk 1]
```diff
@@ -19,8 +19,8 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.WebJobs" Version="3.0.14" />
-    <PackageReference Include="Microsoft.Azure.WebJobs.Host.Storage" Version="3.0.11" />
+    <PackageReference Include="Microsoft.Azure.WebJobs" Version="3.0.15-dev637091944077950638" />
+    <PackageReference Include="Microsoft.Azure.WebJobs.Host.Storage" Version="4.0.0-dev637091944077950638" />
```
Comment (brettsam) [inline src/WebJobs.Extensions/WebJobs.Extensions.csproj]: Can you remind me (I know we talked about it) -- what does this depend on from Host.Storage?

Comment (paulbatum) [reply src/WebJobs.Extensions/WebJobs.Extensions.csproj]: The timer extension lives in this project, and it depends on storage.

**Comment by clauchiorean** [comment]
When do you plan to merge this and have a new version. The current version depends on the old storage extensions (3.0.13)

**Comment by brettsam** [comment]
The only extension dependency here is for our sample app. There's also a *Host* storage package (Microsoft.Azure.WebJobs.Host.Storage) that contains storage-based implementations of a bunch of interfaces. We'll pull that through within the next couple of weeks but there likely won't be much impact on customers as this should mostly be behind-the-scenes.

If you want the latest storage extensions, 4.0.0-preview1 has been released and you should be able to take a dependency on that directly. Or was there something you needed specifically here?

**Comment by clauchiorean** [comment]
> The only extension dependency here is for our sample app. There's also a _Host_ storage package (Microsoft.Azure.WebJobs.Host.Storage) that contains storage-based implementations of a bunch of interfaces. We'll pull that through within the next couple of weeks but there likely won't be much impact on customers as this should mostly be behind-the-scenes.
> 
> If you want the latest storage extensions, 4.0.0-preview1 has been released and you should be able to take a dependency on that directly. Or was there something you needed specifically here?

I was wondering when the official version of all the packages with the new storage libs will be available.
Currently i'm using the source code for now.

**Comment by brettsam** [comment]
Closing and replacing with https://github.com/Azure/azure-webjobs-sdk-extensions/pull/630

**Comment by whitwa** [comment]
> > The only extension dependency here is for our sample app. There's also a _Host_ storage package (Microsoft.Azure.WebJobs.Host.Storage) that contains storage-based implementations of a bunch of interfaces. We'll pull that through within the next couple of weeks but there likely won't be much impact on customers as this should mostly be behind-the-scenes.
> > If you want the latest storage extensions, 4.0.0-preview1 has been released and you should be able to take a dependency on that directly. Or was there something you needed specifically here?
> 
> I was wondering when the official version of all the packages with the new storage libs will be available.
> Currently i'm using the source code for now.

I'm having to use the source code too. When will an updated nuget package be available?

**Comment by brettsam** [comment]
Which source are you using? https://www.nuget.org/packages/Microsoft.Azure.WebJobs.Extensions.Storage has been released for several months now -- 4.0.2 is the latest.

**Pull request closed without merging**